config/config_helpers.go

```diff
@@ -16,12 +17,13 @@ import (
 	"github.com/gruntwork-io/terragrunt/util"
 )
 
-var INTERPOLATION_PARAMETERS = `(\s*"[^"]*?"\s*,?\s*)*`
-var INTERPOLATION_SYNTAX_REGEX = regexp.MustCompile(fmt.Sprintf(`\$\{\s*\w+\(%s\)\s*\}`, INTERPOLATION_PARAMETERS))
+var INTERPOLATION_QUOTED_STRING = `"(?:\\.|[^\\"])*?"`
+var INTERPOLATION_QUOTED_REGEX = regexp.MustCompile(INTERPOLATION_QUOTED_STRING)
+var INTERPOLATION_PARAMETERS = fmt.Sprintf(`\s*(?:%s(?:\s*,\s*%s)*)?\s*`, INTERPOLATION_QUOTED_STRING, INTERPOLATION_QUOTED_STRING)
```

Comment:
With our `INTERPOLATION_QUOTED_STRING` above, we now construct a regex that will accept any number of them in a comma separate list.

The general pattern for comma separated list with at least one element: `A(?:,A)*`

To make it zero or more, we surround the entire thing with a non-capture group and a question mark for zero or one.  This makes our pattern now look like this: `(?:A(?:,A)*)?`

Finally, throw in the `\s*` everywhere, and we have our regex that will match a parameter list with zero or more parameters.

Again, regex101:
![image](https://user-images.githubusercontent.com/766820/59021777-1be08a80-8812-11e9-8204-18c64bd184fb.png)